TFNetworkRecLayer.py

```diff
@@ -5369,10 +5388,7 @@ def get_out_data_from_opts(cls, name, network, add_to_input=False, sources=(), *
     :rtype: Data
     """
     assert len(sources) > 0, "%s %r: must have one source" % (cls, name)
-    if add_to_input:
-      return get_concat_sources_data_template(sources, name="%s_output" % name)  # just the same as the input
-    return super(PositionalEncodingLayer, cls).get_out_data_from_opts(
-      name=name, network=network, sources=sources, **kwargs)
+    return get_concat_sources_data_template(sources, name="%s_output" % name)  # just the same as the input.
```

Comment:
I think this is wrong for `not add_to_input`, or not?

Reply:
When we talked off-line, I thought we said that we could always make the output format the same as the input in all cases (the input was by the way only used to get the time dimension when `add_to_input` was `False`).

Otherwise, some of your suggestions would be also wrong like this one:
`position = tf.reshape(position, [length if i == source_wo_feat.time_dim_axis else 1 for i in range(source_wo_feat.batch_ndim)])`.
This assumes the output shape to be the same as the input, correct?

Reply:
For all the code using `source`, you would just use `self.output` instead, which always has the right format (as defined here).
I think I meant it just for `add_to_input`. For `not add_to_input`, I don't know what makes most sense. Maybe you are right though. Not sure if anyone anyway uses this case.

Reply:
> I think I meant it just for `add_to_input`.

I see. Then let's keep this flexible as before for the `add_to_input` `False` case.